mithril-common/src/signable_builder/cardano_transactions.rs

```diff
@@ -158,29 +158,39 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_compute_merkle_root() {
-        let transaction_1 = CardanoTransaction::new("tx-hash-123", 1, 1);
-        let transaction_2 = CardanoTransaction::new("tx-hash-456", 2, 1);
-        let transaction_3 = CardanoTransaction::new("tx-hash-789", 3, 1);
-        let transaction_4 = CardanoTransaction::new("tx-hash-abc", 4, 1);
-
-        let transactions_set_reference = vec![
-            transaction_1.clone(),
-            transaction_2.clone(),
-            transaction_3.clone(),
-        ];
+    async fn test_compute_merkle_root_in_same_block_range() {
+        let transaction_1 = CardanoTransaction::new("tx-hash-123", 1, 10, "block_hash", 1);
+        let transaction_2 = CardanoTransaction::new("tx-hash-456", 2, 20, "block_hash", 1);
+        let transaction_3 = CardanoTransaction::new("tx-hash-789", 3, 30, "block_hash", 1);
+        let transaction_4 = CardanoTransaction::new("tx-hash-abc", 4, 40, "block_hash", 1);
+
+        for tx in [
+            &transaction_1,
+            &transaction_2,
+            &transaction_3,
+            &transaction_4,
+        ] {
+            assert!(
```

Comment:
Is this assertion necessary? As we are verifying the values that are written directly into the test, not computed.
Would a control in the constructor be relevant?